test/unit/continue_test.lua

```diff
@@ -0,0 +1,242 @@
+local expirationd = require("expirationd")
+local fiber = require("fiber")
+local t = require("luatest")
+local helpers = require("test.helper")
+local g = t.group('expirationd_continue')
+
+local task_name = "walk_all"
+g.before_each(function()
+    g.space = helpers.create_space_with_tree_index('memtx')
+    for _, task in ipairs(expirationd.tasks()) do
+         if task == task_name then
+             expirationd.task(task_name):kill()
+         end
+    end
+end)
+
+g.after_each(function()
+    g.space:drop()
+    if box.space.tmp ~= nil then
+        box.space.tmp:drop()
+    end
+end)
+
+local tuples_wait_event = {{1, "1"}, {2, "2"}, {3, "3"}, {4, "4"}, {5, "5"}}
+local tuples_all = {{1, "1"}, {2, "2"}, {3, "3"}, {4, "4"}, {5, "5"},
+                    {6, "6"}, {7, "7"}, {8, "8"}, {9, "9"}, {10, "10"}}
+local tuples_repeat = {{1, "1"}, {2, "2"}, {3, "3"}, {4, "4"}, {5, "5"},
+                       {1, "1"}, {2, "2"}, {3, "3"}, {4, "4"}, {5, "5"},
+                       {6, "6"}, {7, "7"}, {8, "8"}, {9, "9"}, {10, "10"}}
+
+local function insert_tuples(space)
+    for i = 1,10 do
+        space:insert({i, tostring(i)})
+    end
+end
+
+local function start_walk_task(space, sleep)
+    local cnt = 0
+    local is_expired = function(args, tuple)
+        cnt = cnt + 1
+        if cnt == 6 then
+            if sleep then
+                fiber.sleep(60)
+            else
+                error("test error in iteration")
+            end
+        end
+        return helpers.is_expired_debug(args, tuple)
+    end
+    local task = expirationd.start(task_name, space.id, is_expired,
+                                   {process_expired_tuple = function() return true end})
+    return task
+end
+
+function g.test_task_continue_after_error()
```

Comment:
So what was the expected behavior before this patch? `tuples_repeat` in the second assert?

Reply:
Yes, it is.